benchmarks/piazza/piazza.rs

```diff
@@ -0,0 +1,332 @@
+    let fanout = args.value_of("fanout").unwrap();
+    let csv = args.is_present("csv");
+
+    let class_putter = app.soup.get_mutator(app.class);
+    let user_putter = app.soup.get_mutator(app.user);
+    let taking_putter = app.soup.get_mutator(app.taking);
+    let post_putter = app.soup.get_mutator(app.post);
+
+    println!("Seeding...", );
+    populate_users(nusers, user_putter);
+    populate_classes(nclasses, class_putter);
+    populate_taking(nclasses, nusers, taking_putter, fanout);
+
+    match benchmark.as_ref() {
+        "migration" => {
+            for pid in 0..nposts {
+                post_putter.put(vec![
+                    pid.into(),
+                    (pid % nclasses).into(),
+                    (pid % nusers).into(),
+                    "post".into()
+                    ]);
+            }
+        },
+        "write" => {
+
+        }
+        _ => {
+            println!("wrong benchmark!");
+            return
+        }
+    }
+
+    println!("Finished seeding! Sleeping for 1 second...");
+    thread::sleep(time::Duration::from_millis(1000));
+
+
+    let mut times = Vec::new();
+    if csv {
+       File::create("out.csv").unwrap();
+    }
+
+    println!("Starting benchmark...");
+    for uid in 0..(nusers) {
+        let start;
+        let end;
+
+        match benchmark.as_ref() {
+            "migration" => {
+                start = time::Instant::now();
+                app.log_user(uid.into(), domain_config);
+
+                end = time::Instant::now().duration_since(start);
+            },
+            "write" => {
+                start = time::Instant::now();
+                for i in 0..1000 {
+                    post_putter.put(vec![i.into(), (i % nclasses).into(), (i % nusers).into(), "post".into()]);
+                }
+                end = time::Instant::now().duration_since(start);
+
+                thread::sleep(time::Duration::from_millis(1000));
+
+                app.log_user(uid.into(), domain_config);
+            },
+            _ => {
+                println!("wrong benchmark!");
+                return
+            }
+        };
+
+        let time = (end.as_secs() as f64) +
+                       (end.subsec_nanos() as f64 / 1_000_000_000.0);
+
+        times.push(time);
+
+        if csv {
+            let mut f = OpenOptions::new().write(true).append(true).open("out.csv").unwrap();
+            writeln!(f, "{:?},{:?}", uid, time).unwrap();
+        } else {
+            println!("{:?}: {:?}", uid, time);
+        }
+    }
+
+    println!("{:?} results ", benchmark);
+    println!("avg: {:?}", avg(&times) );
+    println!("max: {:?}", max_duration(&times) );
+    println!("min: {:?}", min_duration(&times) );
+
+    println!("Done with benchmark.");
+
+}
+
+fn max_duration(stats: &Vec<f64>) -> f64 {
+    let max = stats.iter().fold(0f64, |acc, el| {
+        f64::max(acc, *el)
+    });
+
+    max
+}
+
+fn min_duration(stats: &Vec<f64>) -> f64 {
```

Comment:
Ohhhh. Ugh. Damn NaN, eh @ms705? Okay, I guess we'll just have to leave it the way it is for now.